### PR TITLE
Made HttpMethod match case-INsensitive

### PR DESCRIPTION
**Pull request by kichalla**
Related to https://github.com/aspnet/CORS/issues/77

### CODE REVIEW

**Comment by rynowak** [comment]
tests?

File: test/Microsoft.AspNetCore.Mvc.Cors.Test/CorsAuthorizationFilterTest.cs [hunk 1]
```diff
@@ -20,8 +20,10 @@ namespace Microsoft.AspNetCore.Mvc.Cors
 {
     public class CorsAuthorizationFilterTest
     {
-        [Fact]
-        public async Task PreFlightRequest_SuccessfulMatch_WritesHeaders()
+        [Theory]
+        [InlineData("Options")]
+        [InlineData("OPTIONS")]
+        public async Task CaseInsensitive_PreFlightRequest_SuccessfulMatch_WritesHeaders(string preflightRequestMethod)
```
Comment (javiercn) [inline test/Microsoft.AspNetCore.Mvc.Cors.Test/CorsAuthorizationFilterTest.cs]: nit: one extra with all lowercase

**Comment by javiercn** [comment]
:shipit:

**Comment by kichalla** [comment]
:up: :date:

File: src/Microsoft.AspNetCore.Mvc.Cors/Internal/DisableCorsAuthorizationFilter.cs [hunk 2]
```diff
@@ -40,7 +40,7 @@ public Task OnAuthorizationAsync(AuthorizationFilterContext context)
             if (string.Equals(
                     context.HttpContext.Request.Method,
                     CorsConstants.PreflightHttpMethod,
-                    StringComparison.Ordinal) &&
+                    StringComparison.OrdinalIgnoreCase) &&
                 !StringValues.IsNullOrEmpty(accessControlRequestMethod))
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Cors/Internal/DisableCorsAuthorizationFilter.cs]: This class needs a new test also

**Comment by kichalla** [comment]
:up: :date:

**Comment by rynowak** [comment]
:shipit:

**Comment by kichalla** [comment]
936f5a4f6a89d52bd7c2ea0203d20bbdbd0306a1

**Pull request closed without merging**